src/test/munit/oracle/arrays-and-structs-xa-oracle-test-case.xml

```diff
@@ -0,0 +1,118 @@
+                </on-error-continue>
+            </error-handler>
+        </try>
+        <try>
+            <db:execute-script config-ref="${config}">
+                <db:sql><![CDATA[DROP TABLE  TEST_LISTENER_EXTERNAL_CONNECTION]]></db:sql>
+            </db:execute-script>
+            <error-handler>
+                <on-error-continue enableNotifications="true" logException="false">
+                </on-error-continue>
+            </error-handler>
+        </try>
+        <db:execute-ddl config-ref="${config}">
+            <db:sql><![CDATA[CREATE TABLE TEST_XA_TRANSACTION (id number)]]></db:sql>
+        </db:execute-ddl>
+        <db:execute-ddl config-ref="${config}">
+            <db:sql><![CDATA[CREATE TABLE TEST_LISTENER_EXTERNAL_CONNECTION (id number)]]></db:sql>
+        </db:execute-ddl>
+        <db:execute-ddl config-ref="${config}">
+            <db:sql>create or replace type TEST_STRUCT as object (field1 varchar2(3), field2 number(2), field3 varchar2(60))</db:sql>
+        </db:execute-ddl>
+        <db:execute-ddl config-ref="${config}">
+            <db:sql>create or replace type TEST_STRUCT_COLLECTION as table of TEST_STRUCT</db:sql>
+        </db:execute-ddl>
+        <db:execute-ddl config-ref="${config}">
+            <db:sql><![CDATA[
+create or replace procedure retain_first_struct (z in out TEST_STRUCT_COLLECTION)
+    IS
+begin
+    if z is not null and z.count > 1 then
+        z := TEST_STRUCT_COLLECTION(z(1));
+    end if;
+end;
+]]></db:sql>
+        </db:execute-ddl>
+    </munit:before-suite>
+
+    <munit:test name="prepareArrayAndStructLocalXATransaction" ignore="#[java!org::mule::extension::db::DbMunitUtils::isTestIgnored('oracle')]">
+        <munit:execution>
+            <try transactionType="XA">
+                <db:insert  config-ref="${config}">
+                    <db:sql ><![CDATA[INSERT INTO TEST_XA_TRANSACTION VALUES(1)]]></db:sql>
+                </db:insert>
+                <set-variable value="#[Db::prepareArray('TEST_STRUCT_COLLECTION', [ Db::prepareStruct('TEST_STRUCT', ['hi', 1, 'bye']), Db::prepareStruct('TEST_STRUCT', ['hi2', 2, 'bye2']) ])]" variableName="list" />
+                <db:stored-procedure  config-ref="${config}" inOutParameters='#[{ "list": vars.list }]'>
+                    <db:sql><![CDATA[call retain_first_struct(:list)]]></db:sql>
+                </db:stored-procedure>
+            </try>
+        </munit:execution>
+        <munit:validation>
+            <munit-tools:assert-equals actual="#[payload.list[0][1]]" expected="#[1]"/>
+        </munit:validation>
+    </munit:test>
+
+    <munit:test name="prepareArrayAndStructLocalXATransactionExternalConnection" ignore="#[java!org::mule::extension::db::DbMunitUtils::isTestIgnored('oracle')]">
+        <munit:execution>
+            <try transactionType="XA">
+                <db:insert config-ref="oracleDbPoolingListenerExternalConfig">
```

Comment:
Please could you move the external operation after the set variable operation? Because what we want to test is that the prepareArray does not malfunction when mixing connections inside and outside the transaction. (Even, I think is accurate that the external operation also uses the prepareArray)